src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+ */
+@CommandLineProgramProperties(summary="Find reads that evidence breakpoints.",
+        oneLineSummary="Dump FASTQs for local assembly of putative genomic breakpoints.",
+        programGroup = SparkProgramGroup.class)
+public final class FindBreakpointEvidenceSpark extends GATKSparkTool {
+    private static final long serialVersionUID = 1L;
+    private static final int MIN_MAPQ = 20;
+    private static final int MIN_MATCH_LEN = 45; // minimum length of matched portion of an interesting alignment
+    //private static final int MAX_FRAGMENT_LEN = 2000;
+    private static final int MAX_COVERAGE = 1000;
+    private static final float ASSEMBLY_TO_MAPPED_SIZE_RATIO = 7.f;
+    private static final int MIN_HF_KMER_COUNT = 200;
+
+    private static final SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss ");
+
+    @Argument(doc = "directory for fastq output", shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    private String outputDir;
+
+    @Argument(doc = "directory for evidence output", fullName = "breakpointEvidenceDir", optional = true)
+    private String evidenceDir;
+
+    @Argument(doc = "file for breakpoint intervals output", fullName = "breakpointIntervals", optional = true)
+    private String intervalFile;
+
+    @Argument(doc = "file for mapped qname intervals output", fullName = "qnameIntervalsMapped", optional = true)
+    private String qNamesMappedFile;
+
+    @Argument(doc = "file for kmer intervals output", fullName = "kmerIntervals", optional = true)
+    private String kmerFile;
+
+    @Argument(doc = "file for mapped qname intervals output", fullName = "qnameIntervalsForAssembly", optional = true)
+    private String qNamesAssemblyFile;
+
+    /**
+     * This is a path to a file of kmers that appear too frequently in the reference to be usable as probes to localize
+     * reads.  We don't calculate it here, because it depends only on the reference.
+     * The program FindBadGenomicKmersSpark can produce such a list for you.
+     */
+    @Argument(doc = "file containing ubiquitous kmer list", fullName = "kmersToIgnore")
+    private String kmersToIgnoreFilename;
+
+    @Override
+    public boolean requiresReads()
+    {
+        return true;
+    }
+
+    @Override
+    protected void runTool( final JavaSparkContext ctx ) {
+        if ( getHeaderForReads().getSortOrder() != SAMFileHeader.SortOrder.coordinate ) {
+            throw new GATKException("The reads must be coordinate sorted.");
+        }
+
+        final Locations locations =
+                new Locations(evidenceDir, intervalFile, qNamesMappedFile, kmerFile, qNamesAssemblyFile);
+        final PipelineOptions pipelineOptions = getAuthenticatedGCSOptions();
+        final SAMFileHeader header = getHeaderForReads();
+        final JavaRDD<GATKRead> reads = getUnfilteredReads();
+        final HopscotchHashSet<QNameAndInterval> qNamesSet;
```

Comment:
A MultiMap collection type would certainly be clearer, and I'll try to pull that together.
The key has to be QName, because when I'm gathering the reads for an interval I know the QName and need to map through to which intervals, if any, the read belongs to.  The natural way would be Map<QName,List<IntervalID>>, but the vast majority of QNames are associated with a single interval, and so having a gazillion lists of 1 element takes much more memory than an occasional repeated QName.